src/views/tickets/JSViewer.js

```diff
@@ -25,48 +25,62 @@ export default class CodeBlock extends Component {
         artifact.filename.endsWith('.js')
       );
 
+      const fileList = [];
+
+      // Loop through artifact filenames, fetch them, push into list
       for (let i in artifacts) {
         const name = artifacts[i].filename;
 
-        if (!this.fileList.find(i => i.key === name)) {
-          getArtifact(ticketId, name)
-            .then(res => {
-              this.fileList.push({
-                key: name,
-                text: <span className="text">{name}</span>,
-                value: beautify(res),
-              });
-            })
-            .catch(error => {
-              this.setState({ filesBeingBlocked: true });
-
-              this.fileList.push({
-                key: name,
+        try {
+          if (!fileList.find(i => i.key === name)) {
+            const res = await getArtifact(ticketId, name);
+            fileList.push({
+              key: i,
+              // text: <span className="text">{name}</span>,
+              text: name,
+              value: beautify(res),
+            });
+          }
+        } catch (error) {
+          this.setState({
+            filesBeingBlocked: true,
+            fileList: [
```

Comment:
Does setting the array here overwrite any other elements in the array? Asking because the previous code used `push` to append. If so, a spread operator might be in order.

Reply:
Good catch, I'll fix that